Apsitvarkom.UnitTests/Api/Controllers/CleaningEventControllerTests.cs

```diff
@@ -293,6 +306,28 @@ public async Task Create_InvalidDataEntered_ValidationResultsInBadRequestRespons
         Assert.That(result.StatusCode, Is.EqualTo(StatusCodes.Status400BadRequest));
         Assert.That(result.Value, Is.Not.Null.And.Not.Empty);
     }
+
+    [Test]
+    public async Task Create_RepositoryThrows_Status500InternalServerErrorReturned()
+    {
+        var createRequest = new CleaningEventCreateRequest
+        {
+            PollutedLocationId = Guid.NewGuid(),
+            StartTime = DateTime.UtcNow.AddDays(1)
+
+        };
+
+        _pollutedLocationRepository.Setup(r => r.GetByPropertyAsync(It.IsAny<Expression<Func<PollutedLocation, bool>>>()))
```

Comment:
you only have a case where `GetByPropertyAsync` throws. however, we have two try/catch blocks there, so we would ideally cover both of them. 

you have to create a new test case, so that everything passes up until the very last step - setup `InsertAsync` to throw and assert that error.

You can have this already existent test to be named `Create_RepositoryThrowsAcquiringPollutedLocation_Status500InternalServerErrorReturned()`
and the new one:
`Create_RepositoryThrowsInsertingCleaningEvent_Status500InternalServerErrorReturned()`